cdap-api/src/main/java/co/cask/cdap/api/Transactional.java

```diff
@@ -34,4 +39,15 @@
    * @throws TransactionFailureException if failed to execute the given {@link TxRunnable} in a transaction
    */
   void execute(TxRunnable runnable) throws TransactionFailureException;
+
+  /**
+   * Executes a set of operations via a {@link TxRunnable} that are committed as a single transaction with a given
+   * timeout. The {@link TxRunnable} can gain access to {@link Dataset} through the provided {@link DatasetContext}.
+   *
+   * @param timeoutInSeconds the transaction timeout for the transaction, in seconds
+   * @param runnable the runnable to be executed in the transaction
+   *
+   * @throws TransactionFailureException if failed to execute the given {@link TxRunnable} in a transaction
+   */
+  void execute(int timeoutInSeconds, TxRunnable runnable) throws TransactionFailureException;
```

Comment:
not sure. I was thinking about that. But because we do not allow long-running transactions, I think we should not impose an upper bound here.